File: src/main/java/no/fnf/secured/ProtectedEndpoint.java
package no.fnf.secured;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;

import static no.fnf.secured.ProtectedEndpoint.SECURED_PATH;

@Path(SECURED_PATH)
public class ProtectedEndpoint {
    static final String SECURED_PATH = "/secured";
    private final String message;

    public ProtectedEndpoint(String message){
        this.message = message;
    }

    @GET
    @Produces({MediaType.APPLICATION_JSON})
    public Response secured() {
        return Response.ok("{ \"message\" : \" " +message+ " \" }").build();
    }
}

File: src/main/kotlin/no/capraconsulting/secured/ProtectedEndpoint.kt
package no.capraconsulting.secured

import no.capraconsulting.secured.ProtectedEndpoint.Companion.SECURED_PATH
import javax.ws.rs.GET
import javax.ws.rs.Path
import javax.ws.rs.Produces
import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response

@Path(SECURED_PATH)
class ProtectedEndpoint(private val message: String) {

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    fun secured(): Response = Response.ok("""{"message": "$message"}""").build()

    companion object {
        internal const val SECURED_PATH = "/secured"
    }
}
